subprojects/gradle/impact/src/main/kotlin/com/avito/impact/plugin/ImpactMetricsSender.kt

```diff
@@ -0,0 +1,48 @@
+package com.avito.impact.plugin
+
+import com.avito.android.sentry.Environment
+import com.avito.android.sentry.EnvironmentInfo
+import com.avito.android.stats.GaugeMetric
+import com.avito.android.stats.StatsDSender
+import com.avito.android.stats.graphiteSeries
+import com.avito.impact.ConfigurationType
+import com.avito.impact.ModifiedProjectsFinder
+import com.avito.impact.configuration.internalModule
+
+class ImpactMetricsSender(
+    private val statsDSender: StatsDSender,
+    private val environmentInfo: EnvironmentInfo
+) {
+
+    init {
+        require(environmentInfo.environment is Environment.CI) {
```

Comment:
Could we decide to create it at the upper level?

Reply:
create what?

Reply:
Create the sender. I want to see all preconditions on the task or plugin level